projects/soft999_ahnaf_shahriar_fw_103/src/main.c

```diff
@@ -0,0 +1,61 @@
+#include "adc.h"
+#include "delay.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+
+static GpioAddress potentiometer_addr = {
+  .port = GPIO_PORT_A,
+  .pin = 6,
+};
+// function to call when the button gets hit
+static void button_interrupt_handler(const GpioAddress *addy, void *context) {
+  uint16_t potentiometer_data = 0;
+  if (adc_read_raw_pin(potentiometer_addr, &potentiometer_data) == STATUS_CODE_OK) {
+    LOG_DEBUG("potentiometer data: %d\n", potentiometer_data);
+  } else {
+    LOG_DEBUG("Failed to read potentiomter data");
+  }
+  delay_ms(200);
+}
+
+int main(void) {
+  interrupt_init();
+  soft_timer_init();
```

Comment:
I don't think soft timer is used in this project, you don't need to include and init it.